core/common/src/main/java/alluxio/security/authentication/SaslStreamServerDriver.java

```diff
@@ -77,9 +77,14 @@ public void onNext(SaslMessage saslMessage) {
       // Respond to client.
       mRequestObserver.onNext(mSaslHandshakeServerHandler.handleSaslMessage(saslMessage));
     } catch (SaslException se) {
+      LOG.debug("Exception while handling SASL message: {}", saslMessage, se);
       mRequestObserver.onError(new UnauthenticatedException(se).toGrpcStatusException());
     } catch (UnauthenticatedException ue) {
+      LOG.debug("Exception while handling SASL message: {}", saslMessage, ue);
       mRequestObserver.onError(ue.toGrpcStatusException());
+    } catch (Exception e) {
+      LOG.debug("Exception while handling SASL message: {}", saslMessage, e);
+      throw e;
```

Comment:
Will this be logged somewhere else since we are throwing it?

Reply:
Yeah, it will be captured by gRPC pipeline and reported to the other side of the connection. It is still useful to log it here so we have easy access to most detailed information.